spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -85,6 +117,19 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
       throw new IllegalArgumentException("Sampling fraction must be larger or equal to 0 in sampling with replacement")
     }
 
+    getRandomSampleByNum((fraction * mDDF.getNumRows).toLong, withReplacement, seed)
+
+  }
+
+  override def getRandomSampleApprox(fraction: Double, withReplacement: Boolean, seed: Int): DDF = {
+    if (!withReplacement && (fraction > 1 || fraction < 0)) {
```

Comment:
Here is an analogy: is 1 > 1 a valid expression for subsetting a DDF? Do you accept it or say to users that such boolean expression does not make sense?